src/tests/JIT/Regression/JitBlue/Runtime_85765/Runtime_85765.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+public class Runtime_85765
+{
+    public struct S0
+    {
+        public S0(bool f1): this()
+        {
+        }
+    }
+
+    public struct S1
+    {
+        public byte F0;
+        public bool F1;
+        public bool F2;
+    }
+
+    [Fact]
+    public static void Test()
+    {
+        S1 vr2 = M4();
+        vr2.F2 |= vr2.F1;
+        Assert.False(Consume(vr2.F2));
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static S1 M4()
+    {
+        S1 var1 = default(S1);
+        var vr0 = new S0(false);
+        return var1;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool Consume(bool value)
+    {
+        return value;
+    }
+
+    // ------
+
+    [Fact]
+    public static void Test2()
+    {
+        byte* bytes = stackalloc byte[1024];
+        bytes[0x1A] = 1;
+        bytes[0x1B] = 2;
+        int sum = Foo(bytes);
+        Assert.Equal(515, sum);
```

Comment:
I believe that the consensus was not to use Equal due to (1) it relies on reflection and is likely to fail in nativeaot scenarios, and (2) it ends up inlined and makes a mess of the assembly to debug.

Assert.True/False would be ok for (1) but probably not (2).

Reply:
Ok, that's interesting. Can change it to Assert.True/False.